### PR TITLE
 golang: Enable type alias again for go 1.23

### PR DESCRIPTION
**Pull request by sayboras**
All files are re-generated again as there is a change in generation tools.

Relates: https://github.com/kubernetes/kubernetes/pull/127271
Relates: https://github.com/cilium/deepequal-gen/pull/7
Relates: https://github.com/kubernetes-sigs/controller-tools/pull/1061
Relates: 1d9d51f40f3e9ef5d82ec03f8f5a92e2a2697f5d

### CODE REVIEW

**Review by chaunceyjiang — COMMENTED** [review]
image

here

**Comment by sayboras** [comment]
> here

Oops, good catch :+1:

File: go.mod [hunk 1]
```diff
@@ -333,5 +331,6 @@ replace (
 
 	// Using private fork of controller-tools. See commit msg for more context
 	// as to why we are using a private fork.
-	sigs.k8s.io/controller-tools => github.com/cilium/controller-tools v0.16.1-1
+	// For testing purpose, use another fork to include https://github.com/kubernetes-sigs/controller-tools#1061
+	sigs.k8s.io/controller-tools => github.com/sayboras/controller-tools v0.0.0-20240925150353-20695830368c
```
Comment (sayboras) [inline go.mod]: wait for a sync to happen in cilium/controller-tools to pick up https://github.com/kubernetes-sigs/controller-tools/pull/1061

**Review by nathanjsweet — APPROVED** [review]

**Comment by sayboras** [comment]
Convert to draft as it's better to wait for 1.32 RC.

**Review by tklauser — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -119,17 +119,17 @@ require (
...
-	k8s.io/component-base v0.31.2
-	k8s.io/endpointslice v0.31.1
+	k8s.io/api v0.32.0-beta.0
+	k8s.io/apiextensions-apiserver v0.32.0-beta.0
+	k8s.io/apimachinery v0.32.0-beta.0
+	k8s.io/cli-runtime v0.32.0-beta.0
+	k8s.io/client-go v0.32.0-beta.0
+	k8s.io/code-generator v0.32.0-beta.0
+	k8s.io/component-base v0.32.0-beta.0
+	k8s.io/endpointslice v0.32.0-beta.0
```
Comment (aanm) [inline go.mod]: Let's wait until k8s 0.32.0-rc.0 gets released.

**Review by aanm — CHANGES_REQUESTED** [review]

File: go.mod (hunk 2 above)
Comment (sayboras) [reply go.mod]: ah sure, it will be scheduled in 2 weeks IIRC. Marking this as draft.

**Review by aanm — APPROVED** [review]

**Review by brlbil — APPROVED** [review]

**Review by tklauser — APPROVED** [review]

**Pull request merged**